portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -125,6 +123,71 @@ function populateDropdown(list, ID) {
   });
 }
 
+async function placeProximityPins() {
+  const state = document.getElementById('state').value;
+  const userAddress = document.getElementById('address').value;
+  const response = await fetch('/proximity-pins?state=' + state);
+  const pins = await response.json();
+  const radius = Number(document.getElementById('radius').value);
+
+  pins.forEach(async (pin) => {
+    if (await haversineDistance(userAddress, pin.address) < radius) {
+      codeAddress(pin.address);
+    }
+  });
+}
+
+async function haversineDistance(userAddress, dataPoint) {
+  const userAddressQuery = new Promise((resolve, reject) => {
+    geocoder.geocode({'address': userAddress}, function(results, status) {
+      if (status == 'OK') {
+        resolve([
+          results[0].geometry.location.lat(),
+          results[0].geometry.location.lng(),
+        ]);
+      } else {
+        reject(status);
+      }
+    });
+  });
+
+  const pinAddressQuery = new Promise((resolve, reject) => {
+    geocoder.geocode({'address': dataPoint}, function(results, status) {
+      if (status == 'OK') {
+        resolve([
+          results[0].geometry.location.lat(),
+          results[0].geometry.location.lng(),
+        ]);
+      } else {
+        reject(status);
+      }
+    });
+  });
+
+  const [userAddressLatLong, pinAddressLatLong] =
+      await Promise.all([userAddressQuery, pinAddressQuery]);
+
+  const radiusOfEarth = 3958.8;  // Radius of the Earth in miles
```

Comment:
It's good to use a meaningful constant variable for this number instead of relaying on comments.